scality_manila_utils/utils.py

```diff
@@ -0,0 +1,177 @@
+# Copyright (c) 2015 Scality
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import contextlib
+import errno
+import io
+import logging
+import os
+import os.path
+import subprocess
+import tempfile
+
+from scality_manila_utils.exceptions import EnvironmentException
+
+log = logging.getLogger(__name__)
+
+
+@contextlib.contextmanager
+def elevated_privileges():
+    """
+    Obtain temporary root privileges.
+    """
+    previous_uid = os.geteuid()
+    previous_gid = os.getegid()
+    # Become root
+    log.debug("Elevating privileges")
+    os.seteuid(0)
+    os.setegid(0)
+    try:
+        yield
+    finally:
+        # Drop root privileges
+        log.debug("Dropping elevated privileges")
+        os.setegid(previous_gid)
+        os.seteuid(previous_uid)
+
+
+def find_pids(process):
+    """
+    Find pids by inspection of procfs.
+
+    :param process: process name
+    :type process: string
+    :returns: list of pids
+    """
+    process_pids = []
+    pids = filter(lambda f: f.isdigit(), os.listdir('/proc'))
+    for pid in pids:
+        status_path = os.path.join('/proc', pid, 'status')
+        try:
+            with io.open(status_path, 'rt') as f:
+                line = f.readline()
+                _, process_name = line.split()
+                if process_name == process:
+                    process_pids.append(int(pid))
+        except IOError as e:
+            # Pass on processes that no longer exist
+            if e.errno != errno.ENOENT:
+                raise
+
+    log.debug("PIDs for '%s': %s", process, process_pids)
+    return process_pids
+
+
+def binary_check(binary, paths):
```

Comment:
Nothing wrong with this procedure, though I'd let it return a `bool` and let whoever is calling handle the failure case appropriately, without requiring exception handling.